ecosystem/sep-0031.md

```diff
@@ -237,11 +232,19 @@ Name | Type | Description
 
 ##### Customer Info Needed (400 Bad Request)
 
-In the case where the sending anchor didn't provide all the information requested in `/info`, or if the transacton requires extra information, the request should fail with a 400 status code and the following body in JSON format.  The sender should then retry the entire request including all the previously sent fields plus the fields described in the response.
+In the case where the sending anchor didn't provide all the KYC information requested in `GET /customer`, or where the receiver requires additional KYC information after learning of the transaction `amount`, the request should fail with a 400 status code and the following body in JSON format. The sender should then retry both the `GET /customer` request to collect the additional fields and the `PUT /customer` request including all fields described in the `GET /customer` response.
```

Comment:
this handles the case where the sending client just didn't fill out the fields in the first place, but it doesn't handle the case where extra info above the defaults is needed, for example if the amount is large.  See my comment above about passing new types to sep12

Reply:
I think it handles both. The client can pass just a regular `type=sep31-sender` parameter and then have a large `amount` value in the `POST /send` request. The anchor would return this response (400 Customer Info Needed) even though the client has already completed it for a regular sized transaction.

The client can again check the info required from `GET /customer`, which should be changed now that the anchor knows the customer is making a large transaction.

However I agree its way easier to have different `type` values for large transactions that require different KYC. The point is that if they don't, this flow can still work.

Reply:
Maybe we should allow a `choices` field in the `sender_sep12_type` `/info` response.

```
{
    "sender_sep12_type": {
        "description": "A description of this field",
        "choices": ["sep31-sender", "sep31-large-sender"]
    },
    "receiver_sep12_type": {
        "description": "A description of this field",
        "choices": ["sep31-receiver", "sep31-large-receiver"]
    },
    "fields": {...}
}

Reply:
Continuing on this idea ^, I actually think they should be included in the `fields` section in the `/info` response, since they're included there in the `POST /send` request.

```
{
    "fields": {
        "sender_sep12_type": {
            "description": "A description of this field",
            "choices": ["sep31-sender", "sep31-large-sender"]
        },
        "receiver_sep12_type": {
            "description": "A description of this field",
            "choices": ["sep31-receiver", "sep31-large-receiver"]
        },
        "transaction": {...}
    }
}
```

Reply:
Nevermind, the sender won't know which `type` to use.